### PR TITLE
Add quarkus-kotlin plugin

### PR DESCRIPTION
**Pull request by blues-man**
### What does this PR do?
Add language support for Kotlin with Quarkus using VS Code extensions `mathiasfrohlich.kotlin` and `vscode-quarkus`

Kotlin for Quarkus is preview
Kotlin extension for VSCode supports only syntax highlighting

TODO

### CODE REVIEW

File: v3/plugins/blues-man/quarkus-kotlin/0.1.0/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v2
+publisher: blues-man
+name: quarkus-kotlin
+version: 0.1.0
+type: VS Code extension
+displayName: Quarkus Tools with Java 11 and Kotlin
+title: Quarkus Tools for Kotlin
+description: This plug-in provides support for Quarkus development via a Quarkus language server and Kotlin syntax.
+icon: https://raw.githubusercontent.com/JetBrains/logos/master/web/kotlin/kotlin.svg
+repository: https://github.com/blues-man/che-plugin-registry
```
Comment (ericwill) [inline v3/plugins/blues-man/quarkus-kotlin/0.1.0/meta.yaml:10]: This field should match the repository of the `vscode-kotlin` extension.

File: v3/plugins/blues-man/quarkus-kotlin/0.1.0/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v2
+publisher: blues-man
+name: quarkus-kotlin
+version: 0.1.0
```
Comment (ericwill) [inline v3/plugins/blues-man/quarkus-kotlin/0.1.0/meta.yaml:4]: The version should be of the `vscode-kotlin` extension, so 1.7.1 instead 0.1.0.

File: v3/plugins/blues-man/quarkus-kotlin/0.1.0/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v2
+publisher: blues-man
```
Comment (ericwill) [inline v3/plugins/blues-man/quarkus-kotlin/0.1.0/meta.yaml:2]: This should match the publisher field in the [vscode-kotlin package.json](https://github.com/mathiasfrohlich/vscode-kotlin/blob/master/package.json), so in this case `mathiasfrohlich`. The folder structure should also reflect that, i.e. this meta.yaml should be in `v3/plugins/mathiasfrohlich/vscode-kotlin`

File: v3/plugins/blues-man/quarkus-kotlin/0.1.0/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+name: quarkus-kotlin
```
Comment (ericwill) [inline v3/plugins/blues-man/quarkus-kotlin/0.1.0/meta.yaml:3]: This should match the publisher field in the [vscode-kotlin package.json](https://github.com/mathiasfrohlich/vscode-kotlin/blob/master/package.json), so in this case `vscode-kotlin`.

**Review by ericwill — CHANGES_REQUESTED** [review]
Thanks for the PR! I've left a few comments.

Also, could you please add an entry in the [vscode-extensions.json](https://github.com/eclipse/che-plugin-registry/blob/master/vscode-extensions.json) file for this plugin, the entry information should be for the `vscode-kotlin` repository and its related fields. For more info on this file you can look at the [CONTRIBUTE.md file](https://github.com/eclipse/che-plugin-registry/blob/master/CONTRIBUTE.md)

**Comment by blues-man** [comment]
Hi @ericwill thanks for review! I'll fix those and update the PR.

**Comment by blues-man** [comment]
I was thinking that it may better to integrate [this extension](https://github.com/fwcd/vscode-kotlin/i) that supports also the language server, and then to manage this as a Kotlin only plugin (at the moment it contains quarkus ext + kotlin ext). Then in the devfile, adding Quarkus tools plugin + this Kotlin plugin.
There is a little issue with `fwcd` extension activation on Eclipse Che, but I'll open a separate issue for that.

File: v3/plugins/blues-man/quarkus-kotlin/0.1.0/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  containers:
+    - image: "quay.io/eclipse/che-sidecar-java:11-7bd8c8c"
+      name: vscode-quarkus-kotlin
+      memoryLimit: "1500Mi"
+      volumes:
+      - mountPath: "/home/theia/.m2"
+        name: m2
+  extensions:
+    - https://github.com/mathiasfrohlich/vscode-kotlin/releases/download/1.7.1/Kotlin-1.7.1.vsix
+    - https://download.jboss.org/jbosstools/vscode/stable/vscode-quarkus/vscode-quarkus-1.6.0-390.vsix
```
Comment (svor) [inline v3/plugins/blues-man/quarkus-kotlin/0.1.0/meta.yaml:22]: I think you don't need to add vscode-quarkus as it has been already registered in che-plugin-registry  https://github.com/eclipse/che-plugin-registry/blob/master/v3/plugins/redhat/quarkus-java11/1.6.0/meta.yaml.
And in your devfile you'll have to add 2 plugins:  quarkus-java11 and this one (vscode-kotlin)

Comment (blues-man) [reply v3/plugins/blues-man/quarkus-kotlin/0.1.0/meta.yaml:22]: yeah thanks for review @svor, I was commenting proper this, much better to manage this in the devfile, and I would like to use also another extension linked above which supports also the Kotlin language server.

**Comment by blues-man** [comment]
Cloising this and reopening a new one with the `fwcd` extension and new Devfile containing it and existing Quarkus one.

**Comment by sunix** [comment]
I guess this one https://github.com/eclipse/che-plugin-registry/pull/611

**Pull request closed without merging**